telegram-history-dump.rb

```diff
@@ -165,10 +175,20 @@ def process_media(dialog, msg)
     end
     filename = case
       when $config['copy_media']
-        filename = File.basename(response['result'])
-        destination = File.join(get_media_dir(dialog), fix_media_ext(filename))
-        FileUtils.cp(response['result'], destination)
-        destination
+        if response.nil?
+          $log.error('Missing media for message id %s' % msg['id'])
+          "missing_media_%s" % msg['id']
+        else
+          begin
+            filename = File.basename(response['result'])
+            destination = File.join(get_media_dir(dialog), fix_media_ext(filename))
+            FileUtils.cp(response['result'], destination)
+            destination
+          rescue StandardError => e
+            $log.error('Missing media for message id %s' % msg['id'])
+            "missing_media_%s" % msg['id']
+          end
+        end
       else
```

Comment:
You are perfectly right in this case, the code was written... using stackoverflow, if you know what I mean :-)

Feel free to modify it, I'm not really sure to make this work better than you.